azkaban-common/src/test/java/azkaban/trigger/BasicTimeCheckerTest.java

```diff
@@ -77,6 +76,221 @@ public void basicTimerTest() {
     }
 
     assertTrue(cond.isMet());
+  }
+
+  /**
+   * Test Base Cron Functionality.
+   */
+  @Test
+  public void cronTimerTestCurrentZone() {
```

Comment:
Nice test cases.
I see some code duplication here though. Do you think we can extract some of the duplicate pieces out and simplify the test logic?